### PR TITLE
Quote identifiers that contain leading digits or whitespace.

### PR DESCRIPTION
**Pull request by kovrus**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [ ] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: sql-parser/src/main/java/io/crate/sql/Identifiers.java [hunk 1]
```diff
@@ -65,13 +65,8 @@ public static String quoteIfNeeded(String identifier) {
...
     private static boolean areQuotesRequired(String identifier) {
-        for (int i = 0; i < identifier.length(); i++) {
-            int cp = identifier.codePointAt(i);
-            if (cp == '"' || cp == '[' || cp == ']' || cp == '(' || cp == ')' || Character.isUpperCase(cp)) {
-                return true;
-            }
-        }
-        return isKeyWord(identifier);
+        Matcher matcher = IDENTIFIER.matcher(identifier);
+        return isKeyWord(identifier) || !matcher.matches();
```
Comment (mfussenegger) [inline sql-parser/src/main/java/io/crate/sql/Identifiers.java]:
```suggestion
        return isKeyWord(identifier) || !IDENTIFIER.matcher(identifier).matches();
```

To benefit from the short-cuircuiting of `||` and not run the matcher logic if `isKeyWord` is already true.

Comment (kovrus) [reply sql-parser/src/main/java/io/crate/sql/Identifiers.java]: oh, sure!

**Comment by kovrus** [comment]
yes, we should backport to 3.2 and 3.3.

**Comment by kovrus** [comment]
retest this please

**Pull request merged**